gazebo_ros/src/gazebo_ros_force_system.cpp

```diff
@@ -354,18 +354,15 @@ void GazeboRosForceSystemPrivate::ClearLinkWrenches(
   gazebo_msgs::srv::LinkRequest::Request::SharedPtr _req,
   gazebo_msgs::srv::LinkRequest::Response::SharedPtr /*_res*/)
 {
-  bool found = false;
   std::lock_guard<std::mutex> scoped_lock(lock_);
-  for (auto link_wrench_task = link_wrench_tasks_.begin();
-    link_wrench_task != link_wrench_tasks_.end(); ++link_wrench_task)
-  {
-    if ((*link_wrench_task)->link->GetScopedName() == _req->link_name) {
-      link_wrench_tasks_.erase(link_wrench_task--);
-      RCLCPP_INFO(ros_node_->get_logger(), "Deleted wrench on [%s]", _req->link_name.c_str());
-      found = true;
-    }
-  }
-  if (!found) {
+  auto it = std::find_if(link_wrench_tasks_.begin(), link_wrench_tasks_.end(),
+      [_req](auto & link_wrench_task) {
+        return link_wrench_task->link->GetScopedName() == _req->link_name;
+      });
+  if (it != link_wrench_tasks_.end()) {
+    link_wrench_tasks_.erase(it);
+    RCLCPP_INFO(ros_node_->get_logger(), "Deleted wrench on [%s]", _req->link_name.c_str());
+  } else {
```

Comment:
`cppcheck` complained about potential pointer invalidation. Looking at the code, it can be conveniently rewritten with a simple `std::find_if`.

Reply:
I think the new code is slightly different though, because it won't erase multiple occurrences of the same name.